tests/scripts/all.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+import subprocess
+import sys
+from types import FunctionType
+
+BASH_COMPONENT_CODE = '''
+global component_{0}
+def component_{0}():
+    component_proc = subprocess.Popen(['tests/scripts/all.sh', '{0}'], stdout=subprocess.PIPE)
+    for line in component_proc.stdout:
+        line = str(line, 'utf-8')
+        print(line, end='')
+'''
+
+def import_bash_components_file(components_file):
+    # Get a list of functions defined in the components file
+    get_fns_cmd = ['/bin/bash', '-c',
+                   'source {} && compgen -A function'.format(components_file)]
+
+    function_list = subprocess.check_output(get_fns_cmd)
+    function_list = str(function_list, 'utf-8').split()
+
+    # Get the components
+    component_names = filter(lambda x: x.startswith('component_'), function_list)
+    # Remove 'component_' prefix
+    component_names = list(map(lambda x: x[len('component_'):], component_names))
+
+    for component in component_names:
+        # Create a new python function that runs the component in all.sh
+        exec(BASH_COMPONENT_CODE.format(component))
```

Comment:
Please don't use `exec`. Python has good support for data structures and can construct functions dynamically. You don't need `exec` to construct a family of functions parametrized by a component name.